### PR TITLE
Fix for the Safari Startup Crashes

### PR DESCRIPTION
**Pull request by sebv**
Appium now waits specifically for the Apple page instead of the most recent page.

I've also fixed real-safari, no need to press the reload button.
- other minor stuffs.

ping @jlipps @0x1mason

### CODE REVIEW

**Comment by sebv** [comment]
Also tested it in conjunction with #4445

**Comment by 0x1mason** [comment]
This seems awfully fragile.  How confident can we be that the magic `apple` string is a constant in this case?

File: submodules/io.appium.gappium.sampleapp [hunk 1]
```diff
@@ -1 +1 @@
-Subproject commit 3fb5e71b1bc149a57de341b1afe5872c0a2227b9
```
Comment (0x1mason) [inline submodules/io.appium.gappium.sampleapp]: Did you intend to change this?  Doesn't look related.

File: lib/devices/ios/ios.js [hunk 2]
```diff
@@ -1525,23 +1525,43 @@ IOS.prototype.initAutoWebview = function (cb) {
...
-      // the latest window is the one we want.
-      this.navToLatestAvailableWebview(cb);
-    }.bind(this), 6000);
-  } else {
-    this.navToLatestAvailableWebview(cb);
-  }
+IOS.prototype.getContextsAndViews = function (cb) {
+  this.listWebFrames(function (err, webviews) {
+    if (err) return cb(err);
+    var ctxs = [{id:this.NATIVE_WIN}];
```
Comment (jlipps) [inline lib/devices/ios/ios.js]: spacing: `id: this.NATIVE_WIN`

File: lib/devices/ios/ios.js [hunk 3]
```diff
@@ -1525,23 +1525,43 @@ IOS.prototype.initAutoWebview = function (cb) {
...
-    this.navToViewThroughFavorites(cb);
-  } else {
-    this.navToView(cb);
-  }
+IOS.prototype.getLatestWebviewContextForTitle = function (titleRegex, cb) {
+  this.getContextsAndViews(function (err, contexts) {
+    if (err) return cb(err);
+    var matchingCtx;
+    _(contexts).each(function (ctx) {
+      if (ctx.view &&  (ctx.view.title || "").match(titleRegex)) {
```
Comment (jlipps) [inline lib/devices/ios/ios.js]: spacing here after `&&`

File: submodules/io.appium.gappium.sampleapp (hunk 1 above)
Comment (jlipps) [inline submodules/io.appium.gappium.sampleapp]: Yeah, this takes Gappium back a commit, I don't think that's right.

**Comment by jlipps** [comment]
@0x1mason it is a constant in this case because it's built into the Safari app. It will only possibly change when we have a new release of iOS, which requires testing and code updates in Appium anyway; so we'll catch any breaking changes at that point.

I reviewed the code, I think it's a great change and will fix the issue, but there are a few small things that need fixing.

File: submodules/io.appium.gappium.sampleapp (hunk 1 above)
Comment (sebv) [inline submodules/io.appium.gappium.sampleapp]: Yeah that's wrong.

**Comment by sebv** [comment]
Fixed and rebase. 

@0x1mason I've tried on iOS81+iOS61emulators, real devices, a iOS81 emulator with chinese locale. I guess we'll need to keep an eye on this to see if we are missing some use cases.

**Comment by jlipps** [comment]
we only need to run the safari, context, and hybrid tests for this change

**Comment by jlipps** [comment]
and let's not forget to merge #4445 before we merge this.

**Comment by jlipps** [comment]
I think we should merge this and test ASAP.

**Comment by 0x1mason** [comment]
Sounds good.
